# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerJob.cpp
# Commit: fb720435215f
# Full Hash: fb720435215fa5be3b2bd2101f3a54382954b70c
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2024-10-24 09:44:34
# Regressor Bug: 1544232
# File Overlap Count: 4
# Description:
#   Bug 1544232 - Limit lifetime extension of SWs by SWs to the sender's lifetime. r=edenchuang
#   
#   This patch introduces an explicit concept of lifetimes with mechanisms
#   in place so that actions taken by Clients (windows or non-ServiceWorker
#   orkers) will extend the lifetime of a ServiceWorker, but a ServiceWorker
# ==============================================================================

diff -r 25452f6623a0 -r fb720435215f dom/serviceworkers/ServiceWorkerJob.cpp
--- a/dom/serviceworkers/ServiceWorkerJob.cpp	Thu Oct 24 03:02:41 2024 +0000
+++ b/dom/serviceworkers/ServiceWorkerJob.cpp	Thu Oct 24 03:02:42 2024 +0000
@@ -164,10 +164,8 @@
 void ServiceWorkerJob::Finish(ErrorResult& aRv) {
   MOZ_ASSERT(NS_IsMainThread());
 
-  // Avoid double-completion because it can result on operating on cleaned
-  // up data.  This should not happen, though, so also assert to try to
-  // narrow down the causes.
-  MOZ_DIAGNOSTIC_ASSERT(mState == State::Started);
+  // Finishing a job is idempotent and potentially expected by the error
+  // handling path for ServiceWorkerUpdateJob, so this is not an error.
   if (mState != State::Started) {
     return;
   }